### PR TITLE
Add --percentage option for malus node

### PR DESCRIPTION
**Pull request by bredamatt**
This extends the Malus node by allowing for the `--percentage` option when using either `dispute-ancestor`, or `suggest-garbage-candidate` commands, to provide more granularity of control over the behaviour of the Malus node.

The core idea used in the PR is to leverage a Bernoulli distribution to draw a boolean which has a probability `p` of being `true`, decided by the end-user of the Malus node (the probability `p` is the value passed in to the `--percentage` option). This means that malicious behaviour can be adjusted to occur with some probability `p`.

### CODE REVIEW

File: node/malus/src/variants/suggest_garbage_candidate.rs [hunk 1]
```diff
@@ -37,6 +38,8 @@ use polkadot_primitives::v2::{CandidateDescriptor, CandidateHash};
 use polkadot_node_subsystem_util::request_validators;
 use sp_core::traits::SpawnNamed;
 
+use rand::distributions::{Bernoulli, Distribution};
```
Comment (sandreim) [inline node/malus/src/variants/suggest_garbage_candidate.rs:41]: Is it worth to support multiple distributions ?

File: node/malus/src/variants/suggest_garbage_candidate.rs [hunk 2]
```diff
@@ -241,8 +280,30 @@ where
 	}
 }
 
+#[derive(Clone, Debug, clap::Parser)]
+#[clap(rename_all = "kebab-case")]
+#[allow(missing_docs)]
+pub struct SuggestGarbageCandidateOptions {
+	/// Determines the percentage of candidates that should be disputed. Allows for fine-tuning
+	/// the intensity of the behavior of the malicious node. Value must be betweeen 0-100.
+	#[clap(short, long, ignore_case = true, default_value_t = 100)]
```
Comment (sandreim) [inline node/malus/src/variants/suggest_garbage_candidate.rs:289]: We should have some bounds for the value here and fail if value is out of bounds.

File: node/malus/src/variants/suggest_garbage_candidate.rs [hunk 3]
```diff
@@ -241,8 +280,30 @@ where
...
+	pub percentage: u8,
+
+	#[clap(flatten)]
+	pub cmd: RunCmd,
+}
+
 /// Garbage candidate implementation wrapper which implements `OverseerGen` glue.
-pub(crate) struct BackGarbageCandidateWrapper;
+pub(crate) struct BackGarbageCandidateWrapper {
+	/// Pass in the options for the percentage to suggest
```
Comment (sandreim) [inline node/malus/src/variants/suggest_garbage_candidate.rs:298]:
```suggestion
	/// Options from CLI.
```

File: node/malus/src/variants/suggest_garbage_candidate.rs (hunk 2 above)
Comment (bredamatt) [reply node/malus/src/variants/suggest_garbage_candidate.rs]: Sorted with `clap::value_parser`:  https://github.com/paritytech/polkadot/pull/6066/commits/460ba61eb8e9869e9889f8fe52c6e7359d7b9d74

File: node/malus/src/variants/suggest_garbage_candidate.rs (hunk 1 above)
Comment (bredamatt) [reply node/malus/src/variants/suggest_garbage_candidate.rs:41]: In this case the `.sample()` function comes from `Distribution` and not `Bernoulli` so it is necessary.

**Pull request closed without merging**